File: BusReservation/BusAppSolution/BusApp/DTOs/TripDto.cs
using System.ComponentModel.DataAnnotations;

namespace BusApp.DTOs
{
    public class TripDto
    {
        [Required(ErrorMessage = "Bus route ID is required.")]
        public int BusRouteId { get; set; }

        [Required(ErrorMessage = "Bus ID is required.")]
        public int BusId { get; set; }

        [Required(ErrorMessage = "Departure time is required.")]
        public TimeSpan DepartureTime { get; set; }

        [Required(ErrorMessage = "Arrival time is required.")]
        public TimeSpan ArrivalTime { get; set; }

        [Required(ErrorMessage = "Price is required.")]
        [Range(0.1, double.MaxValue, ErrorMessage = "Price must be a positive value.")]
        public decimal Price { get; set; }
    }

}

File: Angular/BusAppFrontend/src/app/core/models/trip.dto.ts
export interface Trip {
    id: number;
    busRouteId: number;
    busId: number;
    departureTime: string; 
    arrivalTime: string;   
    price: number;
  }
  
  export interface NewTrip {
    busRouteId: number;
    busId: number;
    departureTime: string; 
    arrivalTime: string;   
    price: number;
  }
  
  export interface UpdateTrip {
    busRouteId: number;
    busId: number;
    departureTime: string; 
    arrivalTime: string;   
    price: number;
  }